File: src/main/java/com/telmomenezes/synthetic/cli/Const.java
package com.telmomenezes.synthetic.cli;

import com.telmomenezes.synthetic.NetParams;
import com.telmomenezes.synthetic.generators.Generator;
import com.telmomenezes.synthetic.generators.GeneratorFactory;


public class Const extends Command {
    
	@Override
	public String name() {return "const";}
	
	
	@Override
	public String help() {
		String help = "Check if generator weight is constant (random network generator).\n";
		help += "$ synt const -prg <generator>\n";
		help += "Optional parameters:\n";
		help += "-undir if network is undirected.\n";
		help += "-sr <n> sample ratio (default is 0.0006).\n";
		help += "-nodes <n> number of nodes (default is 1000).\n";
		help += "-edges <n> number or edges (default is 10000).\n";
		return help;
    }
	
	
	@Override
	public boolean run() throws SynCliException {
        String progFile = getStringParam("prg");
        double sr = getDoubleParam("sr", 0.0006);
        boolean directed = !paramExists("undir");
        int nodes = getIntegerParam("nodes", 1000);
    	int edges = getIntegerParam("edges", 10000);
        boolean par = paramExists("par");
        String gentype = getStringParam("gentype", "exo");
        	
        NetParams netParams = new NetParams(nodes, edges, directed, par);
        Generator gen = GeneratorFactory.create(gentype, netParams, sr);
        if (gen != null) {
            gen.load(progFile);
            gen.run();
            System.out.println("constant: " + gen.isConstant());
        }
        else {
            System.err.println("could not load program file.");
        }

        return true;
    }
}

File: netgens/commands/const.py
import netgens.generator as gen
from netgens.commands.command import *


class Const(Command):
    def __init__(self, cli_name):
        Command.__init__(self, cli_name)
        self.name = 'const'
        self.description = 'check if generator weight is constant'
        self.mandatory_args = ['prg']
        self.optional_args = ['undir', 'sr', 'nodes', 'edges', 'gentype']

    def run(self, args):
        self.error_msg = None

        prog = args['prg']

        sr = arg_with_default(args, 'sr', DEFAULT_SAMPLE_RATE)
        directed = not args['undir']
        nodes = arg_with_default(args, 'nodes', DEFAULT_NODES)
        edges = arg_with_default(args, 'edges', DEFAULT_EDGES)
        gentype = arg_with_default(args, 'gentype', DEFAULT_GEN_TYPE)

        generator = gen.load(prog, gentype, directed)
        generator.run(nodes, edges, sr)
        print('is constant? %s' % generator.is_constant())

        return True
